File: dtb-api/media-service/src/main/java/app/roaim/dtbazar/mediaservice/domain/ErrorBody.java
package app.roaim.dtbazar.mediaservice.domain;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Value;

import java.time.LocalDateTime;

@Value
public class ErrorBody {

    @JsonProperty("path")
    String path;

    @JsonProperty("requestId")
    String requestId;

    @JsonProperty("error")
    String error;

    @JsonProperty("message")
    String message;

    @JsonProperty("timestamp")
    LocalDateTime timestamp;

    @JsonProperty("status")
    int status;

}

File: dtb-android/app/src/main/java/app/roaim/dtbazar/model/ErrorBody.kt
package app.roaim.dtbazar.model

import com.google.gson.annotations.SerializedName

data class ErrorBody(

	@field:SerializedName("path")
	val path: String? = null,

	@field:SerializedName("requestId")
	val requestId: String? = null,

	@field:SerializedName("error")
	val error: String? = null,

	@field:SerializedName("message")
	val message: String,

	@field:SerializedName("timestamp")
	val timestamp: String? = null,

	@field:SerializedName("status")
	val status: Int? = null
)
